File: tests/clean.pl
#!/usr/bin/env perl -w

use strict;
use Cwd;
use warnings;

my $fileExist = -e "./.git";
my $T = -e "./tests";

if ( $fileExist ) {
    system("git clean -dxf");
    system("git checkout .");
} else {
    if ( $T ) {
        unlink "tests/yarn.lock";
        unlink "tests/shfmt*";
        rmdir "tests/node_modules";
    }
}

File: tests/clean.ts
import * as fs from 'fs';
import { execSync } from 'child_process';

const fileExist = fs.existsSync("./.git");
const T = fs.existsSync("./tests");

if (fileExist) {
    try {
        execSync("git clean -dxf");
        execSync("git checkout .");
    } catch (error) {
        console.error("Error executing git commands:", error);
    }
} else {
    if (T) {
        try {
            fs.unlinkSync("tests/yarn.lock");
            fs.unlinkSync("tests/patch.sh");
            fs.unlinkSync("tests/shfmt");
            fs.rmdirSync("tests/node_modules", { recursive: true });
        } catch (error) {
            console.error("Error handling files or directories:", error);
        }
    }
}
